lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -636,6 +636,16 @@ def write(chunk)
                           record.delete(@trace_key)
                         end
 
+          ts_secs = begin
+                      Integer ts_secs
+                    rescue ArgumentError, TypeError
+                      ts_secs
+                    end
+          ts_nanos = begin
+                       Integer ts_nanos
+                     rescue ArgumentError, TypeError
+                       ts_nanos
```

Comment:
For `nanos`, we can set it to `0` in case of an error, and remove the line down below:

```
ts_nanos = 0 unless ts_nanos.is_a?(Integer)
```